api.js

```diff
@@ -224,3 +229,85 @@ Api.prototype._run = function (files, _options) {
 		return runStatus;
 	});
 };
+
+function getBlankResults() {
+	return {
+		stats: {
+			testCount: 0,
+			passCount: 0,
+			skipCount: 0,
+			todoCount: 0,
+			failCount: 0
+		},
+		tests: []
+	};
+}
+
+Api.prototype._runLimitedPool = function (files, runStatus, concurrency) {
+	var self = this;
+	var tests = {};
+
+	runStatus.on('timeout', function () {
+		Object.keys(tests).forEach(function (file) {
+			var fork = tests[file];
+			fork.exit();
+		});
+	});
+
+	return Promise.map(files, function (file) {
+		try {
+			var test = tests[file] = self._runFile(file, runStatus);
+
+			return new Promise(function (resolve, reject) {
+				var runner = function () {
+					self.emit('ready');
+					var options = {
+						// If we're looking for matches, run every single test process in exclusive-only mode
+						runOnlyExclusive: self.options.match.length > 0
+					};
+					test.run(options)
+						.then(resolve)
+						.catch(reject);
+				};
+
+				test.on('stats', runner);
+				test.on('exit', function () {
+					delete tests[file];
+				});
+				test.catch(runner);
+			}).catch(function (err) {
+				runStatus.handleExceptions({
+					exception: err,
+					file: path.relative('.', file)
+				});
+			});
+		} catch (err) {
+			runStatus.handleExceptions({
+				exception: err,
+				file: path.relative('.', file)
+			});
+		}
+	}, {concurrency: concurrency})
+		.then(function (results) {
+			results = results.filter(function (result) {
+				return Boolean(result); // Filter out undefined results (usually result of caught exceptions)
```

Comment:
I'd be happy with `results.filter(Boolean)`, not sure if there's an official stance on that @sindresorhus?